### PR TITLE
Time Stepping design doc

### PR DESCRIPTION
**Pull request by mark-petersen**
The time stepping design document for Omega-0. During this review, the documentation is [available here](https://portal.nersc.gov/project/e3sm/mpeterse/omega/design-omega0-time-stepping/html/design/TimeStepping.html) for convenient viewing.

Checklist
* [x] Documentation:
  * [x] Documentation has been [built locally](https://e3sm-project.github.io/omega/develop/developers_guide/building_docs.html) and changes look as expected

### CODE REVIEW

File: components/omega/doc/design/TimeStepping.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+### 3.2 Other Schemes
+
+If time permits, other schemes will be added and tested. This includes the four-stage Runge-Kutta (RK4) for direct comparison to MPAS-Ocean. Another possibility is second-order Adams-Bashforth (AB2) which was implemented in MPAS-Ocean. Including multiple time-stepping schemes allows developers to test the modularity requirement above.
+
+## 4 Design
+
+Design specifics will be added at a later time.
+
+## 5 Verification and Testing
```
Comment (sbrus89) [inline components/omega/doc/design/TimeStepping.md:93]: I think we should add a time-only convergence study here.

**Review by sbrus89 — COMMENTED** [review]
@mark-petersen, I think this looks good. I just have one comment about adding a temporal convergence study to the testing section.

File: components/omega/doc/design/TimeStepping.md (hunk 1 above)
Comment (mark-petersen) [reply components/omega/doc/design/TimeStepping.md:93]: I updated it. [See compiled version here](https://portal.nersc.gov/project/e3sm/mpeterse/omega/design-omega0-time-stepping/html/design/TimeStepping.html#verification-and-testing). Let me know what you think.

Comment (sbrus89) [reply components/omega/doc/design/TimeStepping.md:93]: I think that is a good thing to add. We can also do a time convergence test with the manufactured solution where we compare a series of dt refinements on a fixed spatial resolution mesh to a reference solution with a small dt (on the same spatial mesh).

**Comment by mark-petersen** [comment]
Please review when you get a chance: @sbrus89 @mwarusz @hyungyukang 

I would like to make any revisions and merge this week.

**Comment by mwarusz** [comment]
Looks good. I have a comment regarding rhs functions that depend explicitly on time. I remember that in MPAS-O the RK4 scheme had some issues with that. Can we add treating this properly as a required or desired capability ? The description of the forward/backward scheme also doesn't treat this case, but it is needed for the manufactured solution test.

**Review by mwarusz — APPROVED** [review]

**Review by sbrus89 — APPROVED** [review]
Looks good, @mark-petersen. Thanks!

**Review by hyungyukang — APPROVED** [review]
@mark-petersen , Overall, all look great. I left a comment on adding other time stepping methods. We can add more later together after merging.

**Pull request merged**